scripts/solve_network.py

```diff
@@ -145,7 +161,10 @@ def run_lopf(n, allow_warning_status=False, fix_ext_lines=False):
         free_output_series_dataframes(n)
 
         pypsa.opf.network_lopf_build_model(n, formulation=solve_opts['formulation'])
+        
+        add_country_carrier_generation_constraints(n, opts)
         add_opts_constraints(n, opts)
```

Comment:
I think, adding country carrier constraints should be optional and should be activated through a keywords in `snakemake.wildcards.opts`. For example 'CCL'?

```python
if 'CCL' in opts:
  add_country_carrier_generation_constraints(n, opts)
```

Another alternative is to move `add_country_carrier_generation_constraints(n, opts)` into the `add_opts_constraints(n, opts)`.

Reply:
Not entirely shure how to add this in. Please feel free to change my code!

Reply:
I think I second both of Fabian's suggestions, make the inclusion of the constraints conditional on the CCL or some other short acronym, which means you can easily run it with and without the per-country capacity limits:

- `elec_s_12_lv1.25_Co2L-3H-CCL.nc` is then with the per-country-per-carrier constraints, while
- `elec_s_12_lv1.25_Co2L-3H.nc` does not have them

Reply:
Sounds like a good plan to make it opt-in with the config handle.